index.js

```diff
@@ -111,6 +111,28 @@ Babel.prototype.optionsHash = function() {
       hash[key] = (typeof value === 'function') ? (value + '') : value;
     }
 
+    if (options.plugins) {
+      hash.plugins = [];
```

Comment:
We discussed this out of band, hash-for-deps implementation was not clear (likely because it is not well documented). After hash-for-deps value was cleared up, its importance made sense. Yes both techniques should exist, and will work well together.